crates/sequencer_proof_client/src/lib.rs

```diff
@@ -69,51 +76,59 @@ impl TryInto<SnarkProofInputs> for GetSnarkProofPayload {
         }
 
         Ok(SnarkProofInputs {
-            from_block_number: L2BlockNumber(
-                self.block_number_from
+            from_batch_number: L2BatchNumber(
+                self.from_batch_number
                     .try_into()
-                    .expect("block_number_from should fit into L2BlockNumber(u32)"),
+                    .expect("from_batch_number should fit into L2BatchNumber(u32)"),
```

Comment:
maybe return an error instead of expect?

Reply:
We could, but this should never happen and if it does, we'll need to basically blow up the prover, as there's no recovery. I assume blowing earlier is not a big deal. WDYT?